tempo_project/README.md

```diff
@@ -0,0 +1,375 @@
+| `full_state_level`       | PySpark | 1               | Yes                          | Yes                                            | Not Necessary                                       |
+| `state_level_simplified` | PySpark | 1               | Yes                          | Yes                                            | Not Necessary                                       |
+| `simple_profiles`        | PySpark | 1               | Yes                          | Yes                                            | N/A                                                 |
+| `annual_summary_conus`   | PySpark | 1               | Yes                          | Yes                                            | N/A                                                 |
+| `annual_summary_state`   | PySpark | 1               | Yes                          | Yes                                            | N/A                                                 |
+| `annual_summary_county`  | PySpark | 1               | Yes                          | Yes                                            | N/A                                                 |
+
+#### examples-duckdb.ipynb
+
+Dependencies:
+- python>=3.10
+- jupyter
+- duckdb>=0.9.2
+- pandas
+- plotly
+
+Advantages: DuckDB makes the most efficient use of available resources and is trivial to set up.
+
+Limitations: DuckDB is limited to one node and can run out of resources. What datasets you can analyze and what queries you can perform thus depends on the hardware you use.
+
+##### Getting Started
+
+DuckDB easily loads parquet files, including partitioned files assuming the duckdb version is sufficiently recent. For example, a dataset can be loaded and then previewed with the code:
+```python
+import duckdb
+
+def load_table(filepath, tablename):
+    duckdb.sql(f"CREATE TABLE {tablename} AS SELECT * FROM read_parquet('{filepath}/**/*.parquet', hive_partitioning=true, hive_types_autocast=false);")
+    description = duckdb.sql(f"DESCRIBE {tablename};")
+    logger.info(f"Loaded {filepath} as {tablename}:\n{description}")
+
+# load data table
+filepath = data_dir / dataset_name / "table.parquet"
+tablename = "tbl"
+load_table(filepath, tablename)
+duckdb.sql(f"SELECT * FROM {tablename} LIMIT 5").df()
+```
+For example, with `dataset_name = "state_level_simplified"` running this code in the notebook returns:
+![screenshot](docs/duckdb-load-data.png "Notebook output after loading 'state_level_simplified' into DuckDB")
+
+##### Writing Queries
+
+[DuckDB](https://duckdb.org/docs/) provides a variety of interfaces. The example notebook uses the Python package to read .parquet files using the SQL interface. In this case, and assuming the data have been loaded as described above, one example query is:
+```
+df = duckdb.sql(f"""SELECT scenario, {columns_by_type["model_year"]} as year, SUM({value_column})/1.0E6 as annual_twh
```

Comment:
How do you feel about adopting the standard long-form SQL query style like
```
duckdb.sql(f"""
    SELECT scenario
        ,{columns_by_type["model_year"]} as year
        , SUM({value_column})/1.0E6 as annual_twh
    FROM {tablename}
    ...
```
?

Reply:
That's fine with me, although I think I'd prefer the commas to end lines instead of start them.